### PR TITLE
Feature: support more apidoc elements

### PR DESCRIPTION
**Pull request by SteveFlames**
Instead of exposing the custom element `@apiInterface` which creates `@apiSuccess` elements, expose:
`@apiSuccessInterface`
`@apiParamInterface`
`@apiBodyInterface`
`@apiQueryInterface`
so that we can use interfaces for the apidoc elements `@apiSuccess` `@apiParam` `@apiBody` `@apiQuery`

Also, fixed a bug for `Array<...>` syntax when used with namespaces.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -143,19 +155,22 @@ enum PropType {
...
  * Parse element content
  * @param content
+ * @param elementName
  */
-function parse (content: string): ParseResult | null {
+function parse (content: string, elementName: string): ParseResult | null {
   if (content.length === 0) return null
 
-  const parseRegExp = /^(?:\((.+?)\)){0,1}\s*\{(.+?)\}\s*(?:(.+))?/g
+  const parseRegExp = /^(?:\((.+?)\)){0,1}\s*\{(.+?)\}\s*(?:([^\s]*))?\s*(?:(.+))?/g
```
Comment (fofi) [inline src/index.ts:163]: Why this change?

**Review by fofi — APPROVED** [review]
Just one question, super nice job!

File: src/index.ts (hunk 1 above)
Comment (SteveFlames) [reply src/index.ts:163]: to capture the `description` and `field` properties properly (check the docs [here](https://apidocjs.com/#param-api-success) for a reference).

before we were capturing everything after the `{}` block and storing it in the `element` property and as far as I can tell we were actually even misusing it in the `setArrayElements` function.
I think now it's more clear

Comment (SteveFlames) [reply src/index.ts:163]: now that I review it, I should add a default value to `field`.
right now I only do that for arrays (`setArrayElements` function)
